core/app/models/spree/payment_method.rb

```diff
@@ -33,10 +39,23 @@ def payment_source_class
     end
 
     def self.available(display_on = 'both', store: nil)
-      all.select do |p|
-        p.active &&
-          (p.display_on == display_on.to_s || p.display_on.blank?) &&
-          (store.nil? || store.payment_methods.empty? || store.payment_methods.include?(p))
+      Spree::Deprecation.warn "Spree::PaymentMethod.available is deprecated."\
+        "Please use .active, .available_to_users, and .available_to_admin scopes instead."\
+        "For payment methods associated with a specific store, use your_store.payment_methods."
```

Comment:
This isn't a precise replacement ,`your_store.payment_methods` won't return payments if none are associated (it should return all payments in this case). `Payment.available_to_store(store)` which I suggest above would solve this.